api/geyser/src/main/java/org/geysermc/geyser/api/event/lifecycle/GeyserBedrockPing.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2019-2022 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.event.lifecycle;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public record GeyserBedrockPing(String motd1, String motd2, int playerCount, int maxPlayerCount) {
+
+    public static GeyserPingBuilder builder() {
+        return new GeyserPingBuilder();
+    }
+
+    public GeyserPingBuilder toBuilder() {
+        return new GeyserPingBuilder(this);
+    }
+
+    public static class GeyserPingBuilder {
+        private String motd1;
+        private String motd2;
+        private int playerCount;
+        private int maxPlayerCount;
+
+        private GeyserPingBuilder() {
+        }
+
+        public GeyserPingBuilder(GeyserBedrockPing ping) {
+            setMotd1(ping.motd1()).setMotd2(ping.motd2()).setPlayerCount(ping.playerCount()).setMaximumPlayerCount(ping.maxPlayerCount());
+        }
+
+        public GeyserPingBuilder setMotd1(@NonNull String motd1) {
+            this.motd1 = motd1;
+            return this;
+        }
+
+        public GeyserPingBuilder setMotd2(@NonNull String motd2) {
```

Comment:
If arguments are annotated as not null then they should be checked for null and the field should also be annotated